core/src/main/java/nl/nn/adapterframework/compression/ZipWriterPipe.java

```diff
@@ -38,18 +50,73 @@
  */
 public class ZipWriterPipe extends CollectorPipeBase<ZipWriter, MessageZipEntry> {
 
-	private @Getter boolean includeFileHeaders=false;
+	private @Getter boolean includeFileHeaders = false;
+
+	private boolean backwardsCompatibility = false;
+
+	public ZipWriterPipe() {
+		setCollectionName("zipwriterhandle");
+	}
 
 	@Override
 	public void configure() throws ConfigurationException {
-		setCollectionName("zipwriterhandle");
 		super.configure();
 		ZipWriter.validateParametersForAction(getAction(), getParameterList());
 	}
 
 	@Override
-	protected ZipWriter createCollector() throws CollectionException {
-		return new ZipWriter(includeFileHeaders);
+	protected ZipWriter createCollector(Message input, PipeLineSession session) throws CollectionException {
+		String filename = ParameterValueList.getValue(getParameterValueList(input, session), ZipWriter.PARAMETER_FILENAME, "");
+		if(backwardsCompatibility && input.asObject() instanceof String && StringUtils.isEmpty(filename)) {
+			try {
+				filename = input.asString();
+			} catch (IOException e) {
+				throw new CollectionException("unable to convert inputstring to filename");
+			}
+		}
+		return new ZipWriter(includeFileHeaders, filename);
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message input, PipeLineSession session) throws PipeRunException {
+		if(backwardsCompatibility) {
+			try {
+				if(Action.STREAM == getAction()) {
+					return handleStreamAction(input, session);
+				}
+
+				input.preserve();
+				super.doPipe(input, session);
+
+				return new PipeRunResult(getSuccessForward(), input);
+			} catch (IOException e) {
+				throw new PipeRunException(this, "unable to preserve message for action ["+getAction()+"]", e);
+			}
+		}
+
+		return super.doPipe(input, session);
+	}
+
+	private PipeRunResult handleStreamAction(Message input, PipeLineSession session) throws PipeRunException, IOException {
+		try {
+			ParameterValueList pvl = getParameterValueList(input, session);
+			String filename = ParameterValueList.getValue(pvl, ZipWriter.PARAMETER_FILENAME, "");
+			if(StringUtils.isEmpty(filename)) {
+				throw new PipeRunException(this, "filename may not be empty");
+			}
+
+			File collectorsTempFolder = FileUtils.getTempDirectory("collectors");
+			File file = File.createTempFile("msg", ".zip", collectorsTempFolder);
+
+//			doAction(Action.WRITE, PathMessage.asTemporaryMessage(file.toPath()), session);
```

Comment:
I don't like leaving in commented-out code.

Reply:
I agree but it's more to show that is the preferred solution, but then the parameter 'filename' wont resolve anymore 😭